test/evm_test.exs

```diff
@@ -1,8 +1,68 @@
-defmodule EVMTest do
-  use ExUnit.Case
-  doctest EVM
+defmodule EvmTest do
+  use ExUnit.Case, async: true
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  @passing_tests %{
+    arithmetic: [
+      :add0,
+      :add1,
+    ]
+  }
+
+
+  test "Ethereum Common Tests" do
+    for {test_type, test_group} <- @passing_tests do
+      for {test_name, test} <- read_test_file(test_type),
+        Enum.member?(test_group, String.to_atom(test_name)) do
+        db = MerklePatriciaTree.Test.random_ets_db()
+        state = EVM.VM.run(
+          MerklePatriciaTree.Trie.new(db),
+          hex_to_int(test["exec"]["gas"]),
+          %EVM.ExecEnv{
+            machine_code: hex_to_binary(test["exec"]["code"]),
+          }
+        )
+
+        assert_state(test, state)
+        assert elem(state, 1) == hex_to_int(test["gas"])
+      end
+    end
+  end
+
+  def read_test_file(type) do
+    {:ok, body} = File.read(test_file_name(type))
+    Poison.decode!(body)
+  end
+
+  def test_file_name(type) do
+    "test/support/ethereum_common_tests/VMTests/vm#{Macro.camelize(Atom.to_string(type))}Test.json"
   end
+
+  def hex_to_binary(string) do
```

Comment:
This is never very obvious. The answer, I believe, to share a module just for tests is to:

1. Add a file `test/support/encoding_helper.exs` (or whatever) with the functions you have here.
2. Add update `mix.exs` to include files in `test/support` when building in test mode. Jose talks about it (first comment) here: https://stackoverflow.com/a/30652675/320471 and you can see how it's accomplished here: https://github.com/phoenixframework/phoenix/blob/db2f953f24973fab8aa2aab3bd8fc9c87a17adc8/installer/templates/new/mix.exs#L10
3. Either reference `EncodingHelper.hex_to_binary` directly, or add at the time `import EncodingHelper` to get the functions available.

But, this is only really important if you're copying and pasting these functions too many times.